scipy/stats/_bootstrap.py

```diff
@@ -0,0 +1,393 @@
+import numpy as np
+from scipy._lib._util import check_random_state
+from scipy.special import ndtr, ndtri
+from scipy._lib._util import rng_integers
+from dataclasses import make_dataclass
+from ._common import ConfidenceInterval
+
+
+def _jackknife_resample(sample):
+    """Jackknife resample the sample. Only one-sample stats for now."""
+    n = sample.shape[-1]
+
+    # jackknife - each row leaves out one observation
+    j = np.ones((n, n), dtype=bool)
```

Comment:
What happens if someone tries to call this on a series with 200,000 observations?

Reply:
It will raise a memory error. This code is slated to be replaced by Pythran in a separate PR by @jb-leger. Separately, the other memory-intensive parts will optionally be complete in batches.

Reply:
Is this a good idea even in cases where the it will technically fit into memory?  I know from experience (I wrote the [bootstrap in the arch package](https://bashtage.github.io/arch/bootstrap/bootstrap_examples.html)) that there are users who like to bootstrap statistics with millions of observations (e.g., web log data).

Reply:
It depends. If we want to take advantage of vectorization, we want all the bootstrap/jacknife samples at once. I think this concern will be alleviated once the `batch` parameter proposed at the top is implemented.

Reply:
Could you not just implement a simple blocked method, that would loop over blocks when `n` is large? IME you are a better judge of the reasonable batch size than most end users.  I think a blocked approach would be as fast as the current when `n` is smallish but more forgiving as `n` grows large. You would then allocate `(n,block)` arrays rather than `(n,n)`. If `n` was not too big than it would be the case that `(n,block)` would be the same as `(n,n)` and the function would complete in a single iteration.

Reply:
This vectorization does not matter much, as long as the inner loops are vectorized. Its main effect is to eat memory.

Reply:
Yes, I could, but I didn't because a simple blocked methods is a little more complicated than a single function call. There is limited reviewer time - note that this was submitted at the beginning of January and the first comment was at the end of March. So I try to keep things relatively simple for these large PRs. After the initial version is merged, there can be short follow-up PRs that are much easier to review and merge. It's a tough balance to get right, but when there are +2 core dev approvals, I try not to add new features.